src/main/java/org/dinosaur/foodbowl/domain/member/application/MemberService.java

```diff
@@ -37,6 +38,41 @@ public class MemberService {
     private final BookmarkRepository bookmarkRepository;
     private final BlameRepository blameRepository;
 
+    @Transactional(readOnly = true)
+    public MemberProfileResponse getMemberProfile(Long profileMemberId, Long memberId) {
+        Member profileMember = memberRepository.findById(profileMemberId)
+                .orElseThrow(() -> new FoodbowlException(MEMBER_NOT_FOUND));
+        Member member = memberRepository.findById(memberId)
+                .orElseThrow(() -> new FoodbowlException(MEMBER_NOT_FOUND));
+
+        boolean isSelfProfile = profileMember == member;
+
+        return new MemberProfileResponse(
+                profileMember.getNickname(),
+                profileMember.getThumbnailPath(),
+                profileMember.getFollowers().size(),
+                profileMember.getFollowings().size(),
+                isSelfProfile,
+                isFollowed(profileMember, member)
+        );
+    }
+
+    private boolean isFollowed(Member profileMember, Member member) {
+        boolean isFollowed = false;
+        for (Follow follower : profileMember.getFollowers()) {
```

Comment:
```suggestion
        private boolean isFollowed(Member profileMember, Member member) {
        return profileMember.getFollowers()
                .stream()
                .anyMatch(follow -> follow.isFollower(member));
    }
    
    // Follow domain
    public boolean isFollower(Member member) {
        return Objects.equals(follower.getId(), member.getId());
    }
```

follow 도메인에 팔로워와 회원이 일치하는지 확인하는 책임을 주고 확인하는 방법은 어떤지 제안드려봅니다 ㅎㅎ

(팔로워, 팔로잉이 살짝 헷갈리는데 참고만 해주셔도 좋을 것 같습니다 !!)

Reply:
객체지향과 거리가 멀어진 코드를 작성했었네요 🥲 말씀해주신 것처럼 팔로우 객체에 해당 책임을 부여하는 것이 훨씬 좋은 것 같습니다